walk-in-interview/src/main/java/com/google/account/business/servlets/CreateBusinessServlet.java

```diff
@@ -0,0 +1,96 @@
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+/** Servlet that handles creating new business account. */
+@WebServlet("/business-account")
+public final class CreateBusinessServlet extends HttpServlet {
+    private static final long TIMEOUT_SECONDS = 5;
+    private BusinessDatabase businessDatabase;
+
+    @Override
+    public void init() {
+        this.businessDatabase = new BusinessDatabase();
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) {
+        try {
+            Optional<String> optionalUid = FirebaseAuthUtils.getUid(request);
+
+            if (!optionalUid.isPresent()) {
+                System.err.println("Illegal uid");
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                return;
+            }
+
+            String uid = optionalUid.get();
+
+            // Gets business object from the client
+            Business rawBusiness = parseBusinessAccount(request);
+
+            // Forces new account to have empty job list
+            Business business = rawBusiness.toBuilder().setJobs(ImmutableList.of()).build();
+
+            // Stores the account into cloud firestore
+            addBusinessAccount(uid, business);
+
+            // Sends the success status code in the response
+            response.setStatus(HttpServletResponse.SC_OK);
+        } catch (FirebaseAuthException | IOException | ServletException | ExecutionException | TimeoutException e) {
+            // TODO(issue/47): use custom exceptions
+            System.err.println("Error occur: " + e.getCause());
+            // Sends the fail status code in the response
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        }
+    }
+
+    /** Parses into valid Business object from json received from client request. */
+    private Business parseBusinessAccount(HttpServletRequest request) throws IOException, IllegalArgumentException {
+        // Parses job object from the POST request
+        try (BufferedReader bufferedReader = request.getReader()) {
+            String businessAccountJsonStr =
+                    bufferedReader.lines().collect(Collectors.joining(System.lineSeparator())).trim();
+
+            if (StringUtils.isBlank(businessAccountJsonStr)) {
+                throw new IllegalArgumentException("Json for Business object is Empty");
+            }
+
+            Business rawBusiness = ServletUtils.parseFromJsonUsingGson(businessAccountJsonStr, Business.class);
+
+            // Validates the attributes via build()
+            return rawBusiness.toBuilder().build();
+        }
+    }
+
+    /** Adds the business account in the database. */
+    private void addBusinessAccount(String uid, Business business)
+            throws IllegalArgumentException, ServletException, ExecutionException, TimeoutException {
+        try {
+            // Blocks the operation.
```

Comment:
What do you mean by "Blocks the operation" ? Did you mean "This is a blocking operation" ?